dessia_common/datatools/dataset.py

```diff
@@ -666,11 +663,20 @@ def _histogram_unic_value(self, idx_col: int, name_attr: str):
     def _tooltip_attributes(self):
         return self.common_attributes
 
-    def _plot_data_list(self):
-        plot_data_list = []
-        for row, _ in enumerate(self.dessia_objects):
-            plot_data_list.append({attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)})
-        return plot_data_list
+    def _object_to_sample(self, dessia_object: DessiaObject, reference_path: str, row: int):
```

Comment:
It would really be a better code design if object_ to_sample method returned a Sample object, like proposed in my previous comment.

This would : 
- be easier to maintain, because hauling variables through functions and returns is cumbersome,
- thus, be more "Object-Oriented-like" or "pythonic",
- allow a way simpler _to_samples method that could even be written as a one-liner
- globally, be more understandable by any other developper reading this bit of code

Reply:
Done.
Actually it was a little uggly.

Be careful, i write on values attribute of sample in clustereddataset to handle the label of clusters in plot method